extensions/amp-base-carousel/0.1/test-e2e/test-snap-property.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getSlides, getSpacers} from './helpers';
+
+describes.endtoend(
+  'amp-base-carousel test snap property',
+  {
+    testUrl:
+      'http://localhost:8000/test/manual/amp-base-carousel/snap-property.amp.html',
+    environments: ['single'],
+  },
+  async function (env) {
+    let controller;
+
+    beforeEach(async () => {
+      controller = env.controller;
+    });
+
+    it('should have snap property on spacers', async () => {
+      const slides = await getSlides(controller, '#snap-spacers');
+      const spacers = await getSpacers(controller);
+
+      for (let i = 0; i < slides.length; i++) {
+        const slide = slides[i];
+        const styles = await controller.getElementProperty(slide, 'style');
+        await expect(styles).to.not.contain('scroll-snap-align');
+      }
+
+      for (let i = 0; i < spacers.length; i++) {
+        const spacer = spacers[i];
+        const styles = await controller.getElementProperty(spacer, 'style');
+        await expect(styles).to.contain('scroll-snap-align');
+      }
+    });
+
+    it('should have snap property on slides', async () => {
+      const slides = await getSlides(controller, '#snap-slides');
+      for (let i = 0; i < slides.length; i++) {
+        const slide = slides[i];
+        const styles = await controller.getElementProperty(slide, 'style');
+        await expect(styles).to.contain('scroll-snap-align');
+      }
+    });
```

Comment:
1. Let's differentiate by `loop` attribute on selectors, I think that'd be more direct. Feel free to remove ids `#snap-spacers, #snap-slides` from elements as well.

2. Let's split cases so spacers have their own case.

3. I don't think we need to `await` the `expect()` result, but I might be wrong.

```suggestion
    it('should set snap property on spacers', async () => {
      const spacers = await getSpacers(controller);
      for (let i = 0; i < spacers.length; i++) {
        const spacer = spacers[i];
        const styles = await controller.getElementProperty(spacer, 'style');
        expect(styles).to.contain('scroll-snap-align');
      }
    });

    it('should not set snap property on slides when carousel is looped', async () => {
      const slides = await getSlides(controller, 'amp-base-carousel[loop="true"]');
      for (let i = 0; i < slides.length; i++) {
        const slide = slides[i];
        const styles = await controller.getElementProperty(slide, 'style');
        expect(styles).to.not.contain('scroll-snap-align');
      }
    });

    it('should set snap property on slides when carousel is not looped', async () => {
      const slides = await getSlides(controller, 'amp-base-carousel[loop="false"]');
      for (let i = 0; i < slides.length; i++) {
        const slide = slides[i];
        const styles = await controller.getElementProperty(slide, 'style');
        await expect(styles).to.contain('scroll-snap-align');
      }
    });
```

Reply:
SG--FYI all `expect()` in e2e tests must be awaited.